--- C:\Dropbox\Development\[GitHub]\Local\nand2Tetris\nand2Tetris\cmp_files\SquareGame.vm
+++ C:\Dropbox\Development\[GitHub]\Local\nand2Tetris\nand2Tetris\cmp_files\SquareGameCorrect.vm
@@ -1,166 +1,177 @@
	// class SquareGame {
	//     field Square square;
	//     field int direction; // 0=none,1=up,2=down,3=left,4=right
	//     constructor SquareGame new() {
	//         let square = Square.new(0, 0, 30);
 function SquareGame.new 0
-push constant 5
+push constant 2
 call Memory.alloc 1
 pop pointer 0
 push constant 0
 push constant 0
 push constant 30
 call Square.new 3
-pop this 3 // FIELD square
+pop this 0

	//         let direction = 0;
 push constant 0
-pop this 4 // FIELD direction
+pop this 1

	//     method void dispose() {
	//         do square.dispose();
 function SquareGame.dispose 0
 push argument 0
 pop pointer 0
+push this 0
-call square.dispose 0
+call Square.dispose 1
 pop temp 0

 	//                 do moveSquare();
+push pointer 0
+call SquareGame.moveSquare 1
 pop temp 0

	//                 do square.decSize();
 if-goto $IF_TRUE18
 goto $IF_FALSE19
 label $IF_TRUE18
+push this 0
-call square.decSize 0
+call Square.decSize 1
 pop temp 0

	//                 do square.incSize();
 if-goto $IF_TRUE20
 goto $IF_FALSE21
 label $IF_TRUE20
+push this 0
-call square.incSize 0
+call Square.incSize 1
 pop temp 0

	//                 let direction = 1;
 if-goto $IF_TRUE22
 goto $IF_FALSE23
 label $IF_TRUE22
 push constant 1
-pop this 4 // FIELD direction

	//             if (key = 133) {
+pop this 1
 label $IF_FALSE23
 push local 0
 push constant 133
 eq

	//                 let direction = 2;
 if-goto $IF_TRUE24
 goto $IF_FALSE25
 label $IF_TRUE24
 push constant 2
-pop this 4 // FIELD direction

	//             if (key = 130) {
+pop this 1
 label $IF_FALSE25
 push local 0
 push constant 130
 eq

	//                 let direction = 3;
 if-goto $IF_TRUE26
 goto $IF_FALSE27
 label $IF_TRUE26
 push constant 3
-pop this 4 // FIELD direction

	//             if (key = 132) {
+pop this 1
 label $IF_FALSE27
 push local 0 // VAR key
 push constant 132
 eq

	//                 let direction = 4;
 if-goto $IF_TRUE28
 goto $IF_FALSE29
 label $IF_TRUE28
 push constant 4
-pop this 4 // FIELD direction

	//             while (~(key = 0)) {
+pop this 1
 label $IF_FALSE29
 label $WHILE_EXP30
 push local 0 // VAR key
 push constant 0
 eq
 not
 not
 if-goto $WHILE_END31

	//                 do moveSquare();
+push pointer 0
+call SquareGame.moveSquare 1
 pop temp 0

	//     method void moveSquare() {
	//         if (direction = 1) {
 function SquareGame.moveSquare 0 // Returns void
 push argument 0
 pop pointer 0
-push this 4 // FIELD direction
+push this 1
 push constant 1
 eq

	//             do square.moveUp();
 if-goto $IF_TRUE32
 goto $IF_FALSE33
 label $IF_TRUE32
+push this 0
-call square.moveUp 0
+call Square.moveUp 1
 pop temp 0

	//         if (direction = 2) {
 label $IF_FALSE33
-push this 4 // FIELD direction
+push this 1
 push constant 2
 eq

	//             do square.moveDown();
 if-goto $IF_TRUE34
 goto $IF_FALSE35
 label $IF_TRUE34
+push this 0
-call square.moveDown 0
+call Square.moveDown 1
 pop temp 0

	//         if (direction = 3) {
 label $IF_FALSE35
-push this 4 // FIELD direction
+push this 1
 push constant 3
 eq

	//             do square.moveLeft();
 if-goto $IF_TRUE36
 goto $IF_FALSE37
 label $IF_TRUE36
+push this 0
-call square.moveLeft 0
+call Square.moveLeft 1
 pop temp 0

	//         if (direction = 4) {
 label $IF_FALSE37
-push this 4 // FIELD direction
+push this 1
 push constant 4
 eq

	//             do square.moveRight();
 if-goto $IF_TRUE38
 goto $IF_FALSE39
 label $IF_TRUE38
+push this 0
-call square.moveRight 0
+call Square.moveRight 1
 pop temp 0